js/cex.js

```diff
@@ -1440,44 +1447,44 @@ module.exports = class cex extends Exchange {
                 type = 'market';
                 orderAmount = baseAmount;
                 cost = quoteAmount;
-                average = orderAmount / cost;
+                average = Precise.stringDiv (orderAmount, cost);
             } else {
-                const ta = this.safeNumber (order, 'ta:' + quoteId, 0);
-                const tta = this.safeNumber (order, 'tta:' + quoteId, 0);
-                const fa = this.safeNumber (order, 'fa:' + quoteId, 0);
-                const tfa = this.safeNumber (order, 'tfa:' + quoteId, 0);
+                const ta = this.safeString (order, 'ta:' + quoteId, 0);
+                const tta = this.safeString (order, 'tta:' + quoteId, 0);
+                const fa = this.safeString (order, 'fa:' + quoteId, 0);
+                const tfa = this.safeString (order, 'tfa:' + quoteId, 0);
                 if (side === 'sell') {
-                    cost = this.sum (this.sum (ta, tta), this.sum (fa, tfa));
+                    cost = Precise.stringAdd (Precise.stringAdd (ta, tta), Precise.stringAdd (fa, tfa));
                 } else {
-                    cost = this.sum (ta, tta) - this.sum (fa, tfa);
+                    cost = Precise.stringSub (Precise.stringAdd (ta, tta), Precise.stringAdd (fa, tfa));
                 }
                 type = 'limit';
                 orderAmount = amount;
-                average = cost / filled;
+                average = Precise.stringDiv (cost, filled);
             }
             const time = this.safeString (order, 'time');
             const lastTxTime = this.safeString (order, 'lastTxTime');
             const timestamp = this.parse8601 (time);
             results.push ({
```

Comment:
@samgermain `safeOrder` expects raw unparsed trades

Reply:
What changes should I make?